code/modules/power/supermatter/supermatter.dm

```diff
@@ -324,6 +324,9 @@
 
 
 /obj/machinery/power/supermatter/Bumped(atom/AM as mob|obj)
+	if(istype(AM,/obj/mecha) || istype(AM,/obj/structure/closet))
+		qdel(AM)
```

Comment:
Should an item have it's own defined Destroy, Qdel will call that before deleting the object, otherwise it's deleted safely.